Add support for new passwordless endpoint

nbandarchi opened the pull request:
### Changes

Fixing `PasswordlessAuthenticator.signIn()`
    
 * Changed grant type to `http://auth0.com/oauth/grant-type/passwordless/otp`
 * Mapping `connection` to `realm` and `password` to `otp`
 * Changing default type from `ro` to `token` in `OAuthAuthenticator.signIn()`
 * Allowing type to be specified in options
    

### References

Passwordless signin process is attempting to use `oauth/ro` endpoint which no longer exists:

- https://support.auth0.com/tickets/00472740

### Testing

- [ ] This change adds unit test coverage
- [ ] This change adds integration test coverage

### Checklist

- [X] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [X] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [X] All existing and new tests complete without errors

davidpatrick reviewed (changes requested):
@nbandarchi thanks for raising the PR and addressing this issue. Unfortunately `oauth/ro` does still exist when a customer has an allow legacy grant type flag enabled on their account.  So in order to accept this PR without cutting a major we will need keep the default behavior of the passwordless `signIn` function.

We can add a deprecation notice to that function and that it will move to `legacySignIn` on the next major.  And in order to handle the functionality of the new endpoint, we can add the `otp` and `realm` parameter and when either is detected switch the other parameters to match the new endpoint.

nbandarchi commented:
Updated so that it's not defaulting to `token`.  if `passwordless.signIn()` is called with `otp` and optionally `realm` it will change the grant type to `http://auth0.com/oauth/grant-type/passwordless/otp` and pass in `token` as an option to `oauth.signIn()`.  Realm will default to `sms` the same way that connection currently does.

davidpatrick reviewed (changes requested):
@nbandarchi thanks for the quick turnaround! Can you please document the `otp` and `realm` parameters in the jsdoc area for that function?  Also we'll need some new tests in `passwordless.tests.js` for `#signIn` to cover the new `otp` usage.

Thanks :pray:

nbandarchi commented:
No problem!  Added tests and documentation.  The tests are just a duplicated and updated copy of the current tests, but still should show that both paths can coexist.

davidpatrick reviewed (approved):
Changes are great! Thanks for the work here
